File: PHP/7php 1/breakChocolate.php
<?php
/*
|--------------------------------------------------------------------------
| Breaking chocolate problem
|--------------------------------------------------------------------------
|
*/

function breakChocolate ($n, $m) {
   return max($n * $m - 1, 0);
}


function breakChocolate (int $n, int $m) :int {
   return $n * $m - 1;
}


function breakChocolate ($n, $m) {
   return ($n - 1) + ($m - 1) * $n;
}


print_r(breakChocolate(5, 5) . ' ' . 24 . '
');
print_r(breakChocolate(7, 4) . ' ' . 27 . '
');
print_r(breakChocolate(1, 1) . ' ' . 0 . '
');

File: Python/7py 1/break_chocolate.py
"""
|--------------------------------------------------------------------------
| Breaking chocolate problem
|--------------------------------------------------------------------------
|
"""

def break_chocolate(n, m):
   return max(n*m-1, 0)


def breakChocolate(n, m):
   return 0 if n*m==0 else n*m-1


def breakChocolate(n, m):
   return n*m-1 if n and m else 0


print(break_chocolate(5, 5), 24)
print(break_chocolate(7, 4), 27)
print(break_chocolate(1, 1), 0)